src/Apitizer/JsonApi/ResourceObject.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+namespace Apitizer\JsonApi;
+
+class ResourceObject
+{
+
+    /** @var array */
+    protected $attributes = [];
+
+    /** @var array */
+    protected $relationships;
+
+    /** @var string  */
+    protected $type;
+
+    /** @var string  */
+    protected $id;
+
+    public function __construct(string $type, string $id)
+    {
+        $this->type = $type;
+        $this->id = $id;
+    }
+
+    /**
+     * @param  array   $attributes
+     * @param  string  $type
+     * @param  string  $id
+     * @return ResourceObject
+     */
+    public static function factory(array $attributes, string $type, string $id)
+    {
+        $resourceObject = new self($type, $id);
```

Comment:
Use the contstructor? Or `factory(ResourceObject::class)->create($attributes);`

Reply:
je can use the factory pattern also just for initing your object if you have some specific create logic you dont always want inside your constructor. I remember we had this discussion before ;)

Reply:
We didn't finish that discussion